amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/rm/rest/api/impl/RMRolesImpl.java

```diff
@@ -0,0 +1,264 @@
+    }
+
+    private Set<Role> getRolesByFilter(NodeRef filePlan, RolesFilter rolesFilter)
+    {
+        if (rolesFilter.getPersonId() != null)
+        {
+            return filePlanRoleService.getRolesByUser(filePlan, rolesFilter.getPersonId(), rolesFilter.includeSystemRoles());
+        }
+        else
+        {
+            return filePlanRoleService.getRoles(filePlan, rolesFilter.includeSystemRoles());
+        }
+    }
+
+    private RoleModel createRoleModel(NodeRef filePlan, Role role, List<String> include)
+    {
+        List<String> assignedUsers = getAssignedUsers(filePlan, role, include);
+        List<String> assignedGroups = getAssignedGroups(filePlan, role, include);
+
+        return nodesModelFactory.createRoleModel(role, assignedUsers, assignedGroups);
+    }
+
+    private List<String> getAssignedUsers(NodeRef filePlan, Role role, List<String> include)
+    {
+        if (include != null && include.contains(PARAM_INCLUDE_ASSIGNED_USERS))
+        {
+            return new ArrayList<>(filePlanRoleService.getAllAssignedToRole(filePlan, role.getName()));
+        }
+        return null;
+    }
+
+    private List<String> getAssignedGroups(NodeRef filePlan, Role role, List<String> include)
+    {
+        if (include != null && include.contains(PARAM_INCLUDE_ASSIGNED_GROUPS))
+        {
+            return new ArrayList<>(filePlanRoleService.getGroupsAssignedToRole(filePlan, role.getName()));
+        }
+        return null;
+    }
+
+    public void setNodesModelFactory(ApiNodesModelFactory nodesModelFactory)
+    {
+        this.nodesModelFactory = nodesModelFactory;
+    }
+
+    public void setFilePlanRoleService(FilePlanRoleService filePlanRoleService)
+    {
+        this.filePlanRoleService = filePlanRoleService;
+    }
+
+    private RolesFilter getRolesFilter(Query queryParameters)
+    {
+        var rolesFilterBuilder = RolesFilter.builder();
+
+        if (queryParameters != null)
+        {
+            var propertyWalker = new RolesQueryWalker();
+            QueryHelper.walk(queryParameters, propertyWalker);
+
+            rolesFilterBuilder
+                    .withPersonId(propertyWalker.getPersonId())
+                    .withCapabilities(propertyWalker.getCapabilitiesNames())
+                    .withIncludeSystemRoles(propertyWalker.includeSystemRoles());
+        }
+        return rolesFilterBuilder.build();
+    }
+
+    private static class RolesQueryWalker extends MapBasedQueryWalker
+    {
+        private List<String> capabilitiesNames;
+
+        public RolesQueryWalker()
+        {
+            super(LIST_ROLES_QUERY_PROPERTIES, null);
+        }
+
+        @Override
+        public void in(String propertyName, boolean negated, String... propertyValues)
+        {
+            if (negated)
+            {
+                throw new InvalidArgumentException("Cannot use NOT for " + propertyName);
+            }
+
+            if (propertyName.equalsIgnoreCase(PARAM_CAPABILITY_NAME))
```

Comment:
is propertyName will always contains some value ? 

Otherwise we can reverse the logic for null safety
```suggestion
            if (PARAM_CAPABILITY_NAME.equalsIgnoreCase(propertyName))
```